core/transaction_pool.go

```diff
@@ -180,30 +180,28 @@ func (pool *TxPool) SetLocal(tx *types.Transaction) {
 // validateTx checks whether a transaction is valid according
 // to the consensus rules.
 func (pool *TxPool) validateTx(tx *types.Transaction) error {
-	// Validate sender
-	var (
-		from common.Address
-		err  error
-	)
-
 	local := pool.localTx.contains(tx.Hash())
 	// Drop transactions under our own minimal accepted gas price
 	if !local && pool.minGasPrice.Cmp(tx.GasPrice()) > 0 {
 		return ErrCheap
 	}
 
-	// Validate the transaction sender and it's sig. Throw
-	// if the from fields is invalid.
-	if from, err = tx.From(); err != nil {
+	currentState, err := pool.currentState()
+	if err != nil {
+		return err
+	}
+
+	// TODO since this can't be checked using block numbers (since this process
+	// is time sensetive rather than block sensetive) we must, for now, use
+	// frontier rule sets. After frontier we must change this to homestead
+	// ruleset.
+	from, err := tx.FromFrontier()
```

Comment:
I'm not totally sure about this. If I understand the pool correctly, this code runs when we "accept" a transaction into the pool for the first time. If so, shouldn't we accept based on block number?

If I already reached the homestead block, I might as well discard all old forms of From, as there's simply no means to incorporate new transactions below that level. A reorg could still happen, but that would need a chain heavier that mine, so that should also be over homestead already.

Accepting all Frontier transactions into the pool post-homestead could open us up to the same transaction pooling issue we fixed in 1.3.3 where non-executable transactions is piled up in the pool's pending area and are constantly transmitted all over the place in the network.